code-studio/src/js/components/progress/peer_review_progress_dot.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import Radium from 'radium';
+import { connect } from 'react-redux';
+
+import { peerReviewShape } from './types';
+import { createOutline, styles } from './progress_dot_styles';
+
+const PeerReviewProgressDot = React.createClass({
+  propTypes: {
+    peerReview: peerReviewShape,
+    courseOverviewPage: React.PropTypes.bool
+  },
+
+  linkText: function () {
+    if (this.props.peerReview) {
+      if (this.props.peerReview.submitted) {
+        return 'Link to your completed review';
+      } else {
+        return 'Link to your in-progress review';
+      }
+    } else {
+      return 'Reviews unavailable at this time';
+    }
+  },
+
+  render() {
+    let icon = this.props.peerReview ? this.props.peerReview.submitted ? 'fa-check' : '' : 'fa-lock';
```

Comment:
Ternary-inside-ternary isn't great — can this be written more clearly with `if`/`else`?